src/repl/scala/tools/nsc/interpreter/IMain.scala

```diff
@@ -136,7 +136,7 @@ class IMain(val settings: Settings, parentClassLoaderOverride: Option[ClassLoade
   // initializes the compiler, returning false if something went wrong
   override def initializeCompiler(): Boolean = global != null
 
-  lazy val global: Global = {
+  lazy val global: Global with ReplGlobal = {
```

Comment:
If you did `object global extends Global with ReplGlobal` we could have the terse `$r.intp.global` type below, which is nice because I always forget how to write it.